Out of bounds error observed during sortition

```
panic: runtime error: index out of range [5] with length 5

goroutine 1746 [running]:
github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user.Provisioners.MemberAt(0xc000e50f00, 0x5, 0x8, 0xc001fe73b0, 0x5, 0x81)
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user/provisioners.go:89 +0x118
github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user.Provisioners.extractCommitteeMember(0xc000e50f00, 0x5, 0x8, 0xc001fe73b0, 0x3c1632b17, 0x20,
0x0, 0x0)
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user/sortition.go:133 +0x9d
github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user.Provisioners.CreateVotingCommittee(0xc000e50f00, 0x5, 0x8, 0xc001fe73b0, 0x7, 0xc0007a8503, 0
x5, 0xc000e50f00, 0x5, 0x8, ...)
        /home/jules/go/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user/sortition.go:108 +0x319
github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user.Provisioners.GenerateCommittees(0xc000e50f00, 0x5, 0x8, 0xc000127cb0, 0x7, 0x200308, 0x5, 0xc
0007a85f8, 0xa38a2f, 0xc0000950c8)
...
```

--- a/pkg/core/consensus/user/provisioners.go
+++ b/pkg/core/consensus/user/provisioners.go
@@ -2,6 +2,7 @@ package user
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
 
 	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/encoding"
@@ -85,9 +86,12 @@ func (p Provisioners) SubsetSizeAt(round uint64) int {
 }
 
 // MemberAt returns the Member at a certain index.
-func (p Provisioners) MemberAt(i int) *Member {
+func (p Provisioners) MemberAt(i int) (*Member, error) {
+	if i > len(p.Set) {
+		return nil, errors.New("index out of bound")
+	}
 	bigI := p.Set[i]
-	return p.Members[string(bigI.Bytes())]
+	return p.Members[string(bigI.Bytes())], nil
 }
 
 // GetMember returns a member of the provisioners from its BLS public key.

--- a/pkg/core/consensus/user/provisioners_test.go
+++ b/pkg/core/consensus/user/provisioners_test.go
@@ -49,7 +49,10 @@ func TestMemberAt(t *testing.T) {
 	sort.Sort(ks)
 
 	for i := 0; i < nr; i++ {
-		m := p.MemberAt(i)
+		m, err := p.MemberAt(i)
+		if !assert.NoError(t, err) {
+			t.FailNow()
+		}
 		assert.True(t, bytes.Equal(m.PublicKeyBLS, ks[i].BLSPubKeyBytes))
 	}
 }

--- a/pkg/core/consensus/user/sortition.go
+++ b/pkg/core/consensus/user/sortition.go
@@ -126,7 +126,7 @@ func (p Provisioners) CreateVotingCommittee(round uint64, step uint8, size int) 
 func (p Provisioners) extractCommitteeMember(score uint64) []byte {
 	for i := 0; ; i++ {
 		// make sure we wrap around the provisioners array
-		if i >= len(p.Members) {
+		if i >= len(p.Set) {
 			i = 0
 		}
 

--- a/pkg/core/consensus/user/sortition.go
+++ b/pkg/core/consensus/user/sortition.go
@@ -124,13 +124,17 @@ func (p Provisioners) CreateVotingCommittee(round uint64, step uint8, size int) 
 // each node's stake from the passed score until we reach zero. The public key
 // of the node that the function ends on will be returned as a hexadecimal string.
 func (p Provisioners) extractCommitteeMember(score uint64) []byte {
+	var m *Member
+	var e error
 	for i := 0; ; i++ {
-		// make sure we wrap around the provisioners array
-		if i >= len(p.Set) {
-			i = 0
+		if m, e = p.MemberAt(i); e != nil {
+			// handling the eventuality of an out of bound error
+			m, e = p.MemberAt(0)
+			if e != nil {
+				log.Panic(e)
+			}
 		}
 
-		m := p.MemberAt(i)
 		stake, err := p.GetStake(m.PublicKeyBLS)
 		if err != nil {
 			// If we get an error from GetStake, it means we either got a public key of a

--- a/pkg/core/consensus/user/sortition_test.go
+++ b/pkg/core/consensus/user/sortition_test.go
@@ -79,8 +79,17 @@ func TestRemoveStake(t *testing.T) {
 	p, _ := consensus.MockProvisioners(10)
 
 	// Let's add two stakes which are not yet active to one of the provisioners.
-	p.MemberAt(2).AddStake(user.Stake{500, 1000, 10000})
-	p.MemberAt(2).AddStake(user.Stake{500, 1000, 10000})
+	m, err := p.MemberAt(2)
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+	m.AddStake(user.Stake{500, 1000, 10000})
+
+	m, err = p.MemberAt(2)
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+	m.AddStake(user.Stake{500, 1000, 10000})
 
 	// Now, extract a committee for round 1 step 1
 	assert.NotPanics(t, func() { p.CreateVotingCommittee(1, 1, 10) })

